docs/_tutorials/read-only-json-api.md

```diff
@@ -0,0 +1,321 @@
+---
+title: Read-Only JSON API
+author: izdwuut
+date: 2020-12-23 10:04:00 +0100
+---
+
+There shouldn't be a necessity to explain what is the [API](https://en.wikipedia.org/wiki/API). I assume that you've landed here to learn it in a context of Jekyll. Let's jump straight to the actual thing, shall we?
+
+A huge inspiration to write this tutorial was a [post](https://forestry.io/blog/build-a-json-api-with-hugo/) written by [Régis Philibert](https://forestry.io/authors/r%C3%A9gis-philibert/). It's about doing the exact same thing in [Hugo](https://gohugo.io/).
+
+I hope that my solution in Jekyll is elegant, too. You just need to define some [layouts]({% link _docs/layouts.md %}) and write a [plugin]({% link _docs/plugins.md %}) to convert your content to JSON. **Don't be afraid if you're not familiar with Ruby** --- the language that Jekyll is written in. It's similar to Python, and you'll get a hang of it quickly.
+
+By the end of the tutorial, you'll expose the following API:
+* `/` --- an index of all posts
+* `/{url}` --- details of a post
+* `/categories` --- an index of categories
+* `/categories/{category}` --- an index of posts under a category
+
+Let's dive in!
+
+---
+
+# Setup
+
+First, you need to go through an [installation]({% link _docs/installation.md %}) process. Having that, you can initialize a new Jekyll project. In your directory of choice, invoke the following:
+
+```sh
+bundle init
```

Comment:
Can you find a tutorial on the site that covers these steps already as a central source of truth? 

It is not nice to duplicate similar but slightly different setup steps across tutorials, as they don't follow a convention and they all have to be updated if a change is needed.

Instead of covering setup steps and things that the user _won't_ need in the Config section, could you just outline what the user _does_ need? 

e.g.

```
_layouts/
  post.html
_posts/
  2020-12-12-hello.html
index.md
_config.yml
```

Then they can use the jekyll new command if they want or create the files by hand.

Reply:
Fixed. You're right. It's better to define something once and reference it whenever possible.